### PR TITLE
feat: Add Redis subscriber for commands

### LINKED ISSUE #374: Implement command API via message bus
Currently MQTT Message bus subscription is not supported within C Device SDK
Its only supporting Publish
This feature will be handy for subscribing to events and then deciding accordingly

### PR DESCRIPTION
**Pull request by iain-anderson**
Closes #374

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/device-sdk-c/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] I am not introducing a breaking change (if you are, flag in conventional commit message with `BREAKING CHANGE:` describing the break)
- [x] I am not introducing a new dependency (add notes below if you are)
- [ ] I have added unit tests for the new feature or bug fix (if not, why?)
- [ ] I have fully tested (add details below) this the new feature or bug fix (if not, why?)
- [ ] I have opened a PR for the related docs change (if not, why?) - existing feature in Go SDK
  <link to docs PR>

## Testing Instructions
Enable MessageBus in the example (Device/UseMessageBus = true, MessageQueue/Type = "redis") and send requests via core-command

### CODE REVIEW

File: src/c/data.h [hunk 1]
```diff
@@ -30,6 +30,8 @@
 #define EX_MQ_CERTFILE "MessageQueue/Optional/CertFile"
 #define EX_MQ_KEYFILE "MessageQueue/Optional/KeyFile"
 #define EX_MQ_SKIPVERIFY "MessageQueue/Optional/SkipCertVerify"
+#define EX_MQ_TOPIC_CMDREQ "MessageQueue/Topics/CommandRequestTopic"
```
Comment (TomBrennan-Eaton) [inline src/c/data.h:33]: Doesn't seem to agree with https://github.com/edgexfoundry/edgex-docs/pull/887/files#diff-548bda31e6e1fbf98c1a7f74eb6f8b3c353e4a2ea5038d7354891ee65cd8f257R123
That PR shows MessageQueue/Internal/Topics/CommandRequestTopic or MessageQueue/External/Topics/CommandRequestTopic

File: src/c/data-mqtt.c [hunk 2]
```diff
@@ -29,6 +34,9 @@ void edgex_mqtt_config_defaults (iot_data_t *allconf)
   iot_data_string_map_add (allconf, EX_MQ_CERTFILE, iot_data_alloc_string ("", IOT_DATA_REF));
   iot_data_string_map_add (allconf, EX_MQ_KEYFILE, iot_data_alloc_string ("", IOT_DATA_REF));
   iot_data_string_map_add (allconf, EX_MQ_SKIPVERIFY, iot_data_alloc_bool (false));
+
+  iot_data_string_map_add (allconf, EX_MQ_TOPIC_CMDREQ, iot_data_alloc_string (reqt, IOT_DATA_TAKE));
+  iot_data_string_map_add (allconf, EX_MQ_TOPIC_CMDRESP, iot_data_alloc_string ("edgex/device/command/response", IOT_DATA_REF));
```
Comment (TomBrennan-Eaton) [inline src/c/data-mqtt.c:39]: Better with a macro defined for edgex/device/command/response ?

File: src/c/data.h (hunk 1 above)
Comment (iain-anderson) [reply src/c/data.h:33]: Hmm, I'm basing this on https://github.com/edgexfoundry/device-sdk-go/blob/main/example/cmd/device-simple/res/configuration.toml
core-command has a different config map than device services?

Comment (TomBrennan-Eaton) [reply src/c/data.h:33]: Ah, yes, I misread the other PR - didn't see the distinction between "Common" and "Ch-Command" configuration sections.  Yours does match the Common Configuration in that PR, as it should.

File: src/c/data-mqtt.c (hunk 2 above)
Comment (iain-anderson) [reply src/c/data-mqtt.c]: yes

**Pull request merged**